features/256-auth-gear-interaction-overview-in-next/api.md

```diff
@@ -0,0 +1,126 @@
+# auth gear interaction overview in next
+
+Base on the new product architecture decision (auth gear + cloud function and drop record gear), skygear next should bring up a new experience on how auth gear is used in cloud function and client SDK.
+
+## Goal
+
+* Keep auth gear simple, focus on auth related functionalities.
+* Keep client SDK simple, less interaction with auth gear directly.
+* More cloud functions, a developer should create corresponding cloud funtion to fulfill application logic.
+
+## Major changes
+
+* Remove admin API in client SDK.
+* Remove user profile support in auth gear.
```

Comment:
I'm thinking, maybe we should consider our use case and becareful if we want to support some very basic "user metadata" instead.

One example is, say when we support the feature of JWT provider, users may want to add some metadata into the claims of JWT tokens, if we have no way for users to define some metadata, there are no easy way for them to include data in their DB into those claims.

Reply:
@chpapa Thanks for your feedback, though I don't think we should consider such complex use case at this moment, in my opinion, even auth gear won't support "user metadata", a developer still can use other JWT provider (not skygear) to meet the requirement.

Reply:
What do you mean the user can still use other JWT provider to meet the requirement? 

For example, if a user has used Skygear Auth to kick start a new app, and later need to use Skygear Auth to integrate with chat, say Layer, which requires First Name, Last Name, Display Name, and Avatar in the JWT claims, how do they do that (without re-implementing everything?)

Reply:
For this case, I don't think developer has to re-implement everything.

Let's think `CF` has `user_profile` table in DB intially, a developer could implement CF to satisfy layer authentication flow easily. According to https://docs.layer.com/reference/client_api/authentication.out, cloud function should have enough information (first name, last name, display name) to generate JWT for layer (`/identity` endpoint in layer's doc) to get session token. 

And in this way, skygear can even create a set of cloud functions to support layer authentication, developer can import layer cloud function set as an plugin.

Reply:
Hmmm, to me it basically meant the developer need to reimplement everything (implement their JWT provider) while Skygear have an auth server.

While Skygear Auth as a JWT provider should be on the roadmap.

Reply:
While we might don't need it in v1, we definitely should make sure the design can accomodate those features on the roadmap.

(It reminds me we may want to take some time to re-organize the feature repo issues to remove those irrelevant in NEXT already. cc: @tensiuyan )